### PR TITLE
Relax the torch version dependency

### PR DESCRIPTION
**Pull request by fabiendupont**
The PyTorch project has released the patch version 2.3.1 with a number of fixes. This change relaxes the version in the requirements files for build and CUDA, in order to allow picking fixes, while not bumping to the next minor version.

### CODE REVIEW

**Comment by dtrifiro** [comment]
`torch==2.3.0` is also a build dependency in `pyproject.toml

**Comment by fabiendupont** [comment]
> `torch==2.3.0` is also a build dependency in `pyproject.toml

Thanks. Commit amended.

**Comment by dtrifiro** [comment]
@fabiendupont this looks good but would you mind applying the same changes to `requirements-cpu.txt`? thanks

**Comment by fabiendupont** [comment]
@dtrifiro, done. Sorry for the delay.

File: requirements-cpu.txt [hunk 1]
```diff
@@ -2,6 +2,6 @@
 -r requirements-common.txt
 
 # Dependencies for x86_64 CPUs
-torch == 2.3.1+cpu; platform_machine != "ppc64le"
+torch >= 2.3.1+cpu, <2.4.0+cpu; platform_machine != "ppc64le"
```
Comment (dtrifiro) [inline requirements-cpu.txt:5]: This looks broken:

You can test this with `env VLLM_CPU_DISABLE_AVX512=true VLLM_TARGET_DEVICE=cpu PIP_EXTRA_INDEX_URL=https://download.pytorch.org/whl/cpu pip install -e . -v`

I think it's enough to specify two torch lines:
```suggestion
torch >= 2.3.1+cpu; platform_machine != "ppc64le"
torch < 2.4.0+cpu; platform_machine != "ppc64le"
```

**Comment by dtrifiro** [comment]
Torch is now set at 2.4.0, so this is not needed anymore.

**Pull request closed without merging**